src/constraint/CollisionFree.cpp

```diff
@@ -27,31 +27,63 @@ statespace::StateSpacePtr CollisionFree::getStateSpace() const
 
 //==============================================================================
 bool CollisionFree::isSatisfied(
-    const aikido::statespace::StateSpace::State* _state) const
+    const aikido::statespace::StateSpace::State* _state,
+    TestableOutcome* outcome) const
 {
   auto skelStatePtr = static_cast<const aikido::statespace::dart::
                                       MetaSkeletonStateSpace::State*>(_state);
   mStatespace->setState(skelStatePtr);
 
   bool collision = false;
   dart::collision::CollisionResult collisionResult;
-  for (auto groups : mGroupsToPairwiseCheck)
+  for (const auto& groups : mGroupsToPairwiseCheck)
   {
     collision = mCollisionDetector->collide(
         groups.first.get(),
         groups.second.get(),
         mCollisionOptions,
         &collisionResult);
+
     if (collision)
+    {
+      if (outcome != nullptr)
+      {
+        const auto& collisionOutcome
```

Comment:
I don't think this should be `const`, since we're changing it below. I'm also unclear as to whether it should be `auto&` or just `auto` -- [this](https://stackoverflow.com/a/29860056) seems to suggest that `auto` is fine?